scripts/imgr

```diff
@@ -1,5 +1,42 @@
 #!/bin/bash
 
+# We need to check hutch early to figure out old vs new ioc manager
+# Add engineering_tools to PATH for get_hutch_name
+DIR=$(readlink -f "$(dirname "${BASH_SOURCE[0]}")")
+PATH=$PATH:$DIR
+
+# Search for hutch in args
+for (( i=1; i<=$#; i++)); do
+    if [[ "${!i}" == "--hutch" ]]; then
+        ARGHUTCH_INDEX=$((i+1))
+        ARGHUTCH=${!ARGHUTCH_INDEX}
+        break
+    fi
+done
+
+# Hutch choice priority order:
+# 1. --hutch arg
+# 2. HUTCH environment variable
+# 3. get_hutch_name
+HUTCH=${ARGHUTCH:-$HUTCH}
+HUTCH=${HUTCH:-$(get_hutch_name)}
+
+: "${PYPS_ROOT:=/cds/group/pcds/pyps}"
```

Comment:
This was new syntax to me.  Is there a reason we chose this way of assigning the variable here specifically?

Reply:
Someone online suggested it and I thought it was neat :shrug:
Perhaps there's a more concise/obvious way to express "If this variable is unset, set it" and I should do that instead.

Reply:
I ended up replacing it with a much simpler construct for readability